components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorGenerator.kt

```diff
@@ -1,14 +1,26 @@
 package io.github.composegears.valkyrie.generator.imagevector
 
 import androidx.compose.material.icons.generator.vector.Vector
+import io.github.composegears.valkyrie.generator.imagevector.OutputFormat.BackingProperty
 
 data class ImageVectorGeneratorConfig(
     val packageName: String,
     val packName: String,
     val nestedPackName: String,
+    val outputFormat: OutputFormat = BackingProperty,
     val generatePreview: Boolean,
 )
 
+enum class OutputFormat {
+    BackingProperty,
+    LazyDelegateProperty,
+    ;
+
+    companion object {
+        fun findValueOf(name: String?): OutputFormat = entries.find { it.name == name } ?: BackingProperty
```

Comment:
```suggestion
        fun from(name: String?): OutputFormat {
            return OutputFormat.valueOf(name ?: "BackingProperty")
        }
```

Reply:
its protection against potential crash
for example some user has preferences and in the plugin version we renamed enum value

Or we can make more safe:

```kotlin
enum class OutputFormat(val key: String) {
    BackingProperty("backing_property"),
    LazyDelegateProperty("lazy_delegate_property"),
    ;

    companion object {
        fun from(key: String?): OutputFormat {
            return OutputFormat.valueOf(key ?: BackingProperty.key)
        }
    }
}
```

Reply:
Yeah, there is no other assignment for the nullable pref, it would be safe to use `valueOf` here, this is also why I suggest declaring `outputFormat` nullable.

Reply:
Updated in last commit